### DISCUSSION TITLE
Possible to add a data component that works similar to localised fields

### DISCUSSION THREAD

**Discussion by markbeazley** [pimcore/pimcore#11423] [Q&A / Support]
I'm investigating using Pimcore as a PIM/MDM for our product data. We have several websites under different brands, some of these sites share products, but with different descriptions and other similar content. I'm trying to figure out the best way to store this data in a single instance of Pimcore, and I don't know if someone has already solved this in the past or if its something pimcore supports out of the box that I've missed.

I'm imagining something that works like the localised fields for translations, so I could define a list of brands, have one be the default, and then allow the fields to be overridden as needed per brand. The other option I have is creating multiple fields per brand, but that doesn't seem to be the best way to handle it for many reasons (more complicated to export and maintain in general).

**Comment by AndrewKMoses** [comment]
There are different options depending on how you want to structure your data.

Inheritance: https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Class_Settings/Inheritance.html

Variants: https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Class_Settings/Variants.html

**Reply by markbeazley** [reply]
Given we will have products that already have variants for size/colour I'm guessing we wouldn't be able to use variants with variants? Looking at it it seems the system does support variants of variants, but I can see that becoming quite a mess as some of our products have over 100 variants, so I'd guess we would need to have a copy of each base variant for each brand.

Inheritance looks promising I think, but I'm guessing that we would if we had one product on 3 different brands have one base data object then 3 brand specific ones that inherit from that? How would it handle unique constraints on say ID fields for example we have a product level ID that is the same across all sites, would we just set that on the base one, and override if needed on the child objects?
